docs/markdown/Python/python-goals/python-package-goal.md

```diff
@@ -139,6 +139,22 @@ You must explicitly add the dependencies you'd like to the `dependencies` field.
 
 This does not work with file arguments; you must use the target address, like `./pants package helloworld:black_bin`.
 
+### Injecting command-line arguments and environment variables
+
+You can use the `inject_args` and `inject_env` fields to "freeze" command-line arguments and environment variables into the PEX file. This can save you from having to create shim files around generic binaries. For example:
+
+```python myproduct/myservice/BUILD
+python_requirement(name="gunicorn", requirements=["gunicorn==20.1.0"])
+
+pex_binary(
+  name="myservice_bin",
+  script="gunicorn",
+  inject_args=["myproduct.myservice.wsgi:app", "--name=myservice"],
+  inject_env={"MY_ENV_VAR=1"},
```

Comment:
I agree with the terminology implementation wise, but from a UX perspective, I believe simply using `arg` and `env` as the field aliases makes more sense as to me in this context I would assume args/env where for the script/entry_point rather than the PEX command to build the pex, so the inject bit here is potentially confusing.

Reply:
Well, this is the terminology in Pex, and also, the purpose of "inject" is to emphasize exactly that these are baked into the pex file, so they are your program's args rather than the pex binary's args.  It may be clear to you, but I'm not sure it would be to others?

Reply:
tl;dr; in summary, I just think it reads better.

I'm not blocking/requesting this change, will just post to help clarify my point of view:

> [...] emphasize exactly that these are baked into the pex file

as opposed to being.. what? -- (only thing that comes to mind would be args/env to the PEX tool when building the pex; easily confused? perhaps)

I am looking at the field names of the target, and have `script` / `entry_point`, `inject_args` and `inject_env`. I don't get the inject part, why isn't the script/entry point also prefixed with inject in that case? I see these fields being the command to invoke when running the pex file, so it's merely breaking down a `gunicorn some:app --name=service`.
If anything, using `script_args`/`script_env` could make sense to make this clear (but doesn't align well with `entry_point`..)

That the inject terminology comes from the PEX tool seems like an implementation detail to me. I would prefer not having to know PEX that well in order to use this target effectively (however referring to this fact in the field docs then could be helpful for those that are familiar with PEX to help make it clear it is the same thing)

I'm fine with go ahead with as-is, just not being "the most obvious" naming to me.. ;)

Reply:
Yeah I didn't think about this too much when writing it, and lifted the names directly from the underlying PEX concepts...

Reply:
Meaning, I can change it with no fuss

Reply:
Done